src/core/app-template.cc

```diff
@@ -107,7 +108,12 @@ const app_template::seastar_options& app_template::options() const {
 app_template::configuration_reader app_template::get_default_configuration_reader() {
     return [this] (bpo::variables_map& configuration) {
         auto home = std::getenv("HOME");
-        if (home) {
+        if (configuration.contains("config-file")) {
+            std::ifstream ifs_io(configuration["config-file"].as<std::string>());
+            if (ifs_io) {
```

Comment:
i think, if user explicitly specifies a `config-file`, but that file is not readable, probably we should fail the whole application instead of swallow the failure and move on. so the user is aware that the option is invalid or something very bad happened.